### ISSUE TITLE
Unexpected size of returned results

### ISSUE THREAD

**Issue by KatharineShapcott** [esi-neuroscience/acme#6]
Hey Stefan,
I've been testing returning the results and it all seemed to be working fine when I had just a few jobs. But I actually need to run 784 computations. So I set up a client with 300 jobs which works fine when write_worker_results=True, I get 784 saved files. But for some reason 300 jobs and 784 inputs gives a result that has a length of 111! 

I'm sure the outputs are all there (presumably now results[0] has a length greater than 1) but I'm not sure what the order of the results should be now. Do you know how dask does this?
Best,
Katharine 

```
n_jobs = 300
client = esi_cluster_setup(partition="8GBXS", n_jobs=n_jobs)

with ParallelMap(comparison_preprocessing, clf_in, data_train, label_train, data_test, n_filters=multifilter_n_filters, n_inputs=n_trys, write_worker_results=False) as pmap:
    results = pmap.compute()

print(len(results))

allpred = np.array([xi[0][0] for xi in results])
times = np.array([xi[0][1] for xi in results])
print(allpred.shape)
print(times.shape)
```

The output looks like this:
```
(acme) [shapcottk@esi-svhpc7 python]$ python filter_net_paper.py 
Calc_nfilters with trainfrac1.00
<esi_cluster_setup> SLURM workers ready: 300/300 elapsed[elapsed time 00:26 | timeout at 03:00]
<esi_cluster_setup> Cluster dashboard accessible at http://10.100.32.7:8787/status
./acme/acme/backend.py:170: RuntimeWarning: <ACMEdaemon> `write_worker_results` is `False` and `taskID` is not a keyword argument of comparison_preprocessing.Results will be collected in memory by caller - this might be slow and can lead to excessive memory consumption. 
  RuntimeWarning, __file__, inspect.currentframe().f_lineno)
<ACMEdaemon> Attaching to global parallel computing client <Client: 'tcp://10.100.32.7:43457' processes=300 threads=300, memory=2.40 TB>
<ACMEdaemon> Preparing 774 parallel calls of `comparison_preprocessing` using 300 workers
<ACMEdaemon> Log information available at /mnt/hpx/slurm/shapcottk/shapcottk_20201120-165036
100% | | 111/111 [06:49<00:00]
<ACMEdaemon> Gathering results in local memory
<ACMEdaemon> SUCCESS! Finished parallel computation. 
111
(111, 10000)
(111,)
```

**Comment by KatharineShapcott** [comment]
Hey Stefan, sorry this took so long to check! I've now tested this issue, works just like I expected now, thanks so much! 👍 

I did find this message a bit confusing when I tested though: 
```
<esi_cluster_setup> Requested job-count 400 exceeds waiter threshold 100: waiting for `n_jobs_startup` jobs to come online, then proceed
```
Becase it's I hadn't set n_jobs_startup myself I didn't realise this was the same as the "waiter threshold". I would change [the message](https://github.com/esi-neuroscience/acme/blob/95403e4632fd878e69c6fdd0ff82f0b787765044/acme/dask_helpers.py#L349-L350) to 
```
msg = "{} Requested job-count {} exceeds `n_jobs_startup` : " +\
            "waiting for {} jobs to come online, then proceed"
```
or similar to make it more clear.
